providers/gce/gce_loadbalancer_internal.go

```diff
@@ -668,7 +668,8 @@ func (g *Cloud) ensureInternalInstanceGroups(name string, nodes []*v1.Node) ([]s
 				parts := strings.Split(ins.Instance, "/")
 				groupInstances.Insert(parts[len(parts)-1])
 			}
-			if names.HasAll(groupInstances.UnsortedList()...) {
+			groupInstanceNames := groupInstances.UnsortedList()
```

Comment:
I'm fine with the code as is, but since this is looking like it will be a long term carry, a comment as to why we have these conditions will help reduce issue archeology in the future.

Reply:
The motivation was included in the commit message. I just brushed up again. 

Do you think that works or we also need an in-code comment?